src/phases/tpch/denormalized/Q1.yml

```diff
@@ -4,31 +4,41 @@ Description: |
   Run TPC-H query 1 against the denormalized schema. Using an 'executionStats' explain causes each command to run its execution plan until no
   documents remain, which ensures that the query executes in its entirety.
 
-batchSize: &batchSize 101  # The default batch size.
+batchSize: &batchSize {^Parameter: {Name: "BatchSize", Default: 101}}
 query1Delta: &query1Delta {^Parameter: {Name: "Query1Delta", Default: -90}}
 
+TPCHDenormalizedQuery1Aggregation: &TPCHDenormalizedQuery1Aggregation
+  aggregate: customer
+  pipeline:
+    [
+      {$unwind: "$orders"},
+      {$addFields: {lineitem: "$orders.lineitem", "orders.lineitem": "$$REMOVE"}},
+      {$unwind: "$lineitem"},
+      {$replaceWith: "$lineitem"},
+      {$match: {$expr: {$lte: ["$l_shipdate", {$dateAdd: {startDate: {$dateFromString: {dateString: "1998-12-01"}}, unit: "day", amount: *query1Delta}}]}}},
+      {$group: {_id: {l_returnflag: "$l_returnflag", l_linestatus: "$l_linestatus"}, sum_qty: {$sum: "$l_quantity"}, sum_base_price: {$sum: "$l_extendedprice"}, sum_disc_price: {$sum: {$multiply: ["$l_extendedprice", {$subtract: [1, "$l_discount"]}]}}, sum_charge: {$sum: {$multiply: ["$l_extendedprice", {$subtract: [1, "$l_discount"]}, {$add: [1, "$l_tax"]}]}}, avg_qty: {$avg: "$l_quantity"}, avg_price: {$avg: "$l_extendedprice"}, avg_disc: {$avg: "$l_discount"}, count_order: {$count: {}}}},
+      {$project: {_id: 0, l_returnflag: "$_id.l_returnflag", l_linestatus: "$_id.l_linestatus", sum_qty: 1, sum_base_price: 1, sum_disc_price: 1, sum_charge: 1, avg_qty: 1, avg_price: 1, avg_disc: 1, count_order: 1}},
+      {$sort: {l_returnflag: 1, l_linestatus: 1}}
+    ]
+  cursor: {batchSize: *batchSize}
+  allowDiskUse: true
+
 TPCHDenormalizedQuery1:
-  Repeat: 1
-  Database: tpch
+  Repeat: &Repeat {^Parameter: {Name: "Repeat", Default: 1}}
+  Database: &db tpch
+  Operations:
+  - OperationMetricsName: Query1
```

Comment:
**TL;DR:** `TPCHDenormalizedQueryAggregation` is not an operation; it is just some yaml that I am reusing in multiple places. For the actual operation definitions, see `src/workloads/tpch/denormalized/Q1.yaml` in commit 2.

**Longer explanation:**
This is the phase definition which is "loaded" into the workload using `LoadConfig`, and not the workload itself. The operation metrics name is composed with the actor name (e.g. `actor name + operation name`), which is defined in `src/workloads/tpch/denormalized/Q1.yaml` (commit 2).

For query 1 for example (the same pattern applies to every query), we have three actors:
- `TPCHDenormalizedQuery1Cold`
- `TPCHDenormalizedQuery1Warmup`
- `TPCHDenormalizedQuery1Hot`

Each one loads phases differently.
- `TPCHDenormalizedQuery1Cold` loads the phase `TPCHDenormalizedQuery1Explain` which runs with `explain: "executionStats"` and logs the output of the explain.
- `TPCHDenormalizedQuery1Warmup` loads the phase `TPCHDenormalizedQuery1Aggregation` into an operation which simply runs the aggregation once to prepare for the hot run, and does not log the output of the explain. In principle we do not want to track any stats for this stage of the workload (so we exclude OperationMetricsName) but this does not actually stop metrics from being collected for this phase, which is functionality that will be added by the perf team in [TIG-4128](https://jira.mongodb.org/browse/TIG-4128).
- `TPCHDenormalizedQuery1Hot` loads the phase `TPCHDenormalizedQuery1` which is the operation actually used by the hot run. This actually has a metrics name because we do want to collect data about this run, which is why this operation is different from the warmup phase.

Does that all make sense?